### PR TITLE
Add experimental CalendarDatePicker component

### PR DESCRIPTION
**Pull request by Konyaco**
image

### CODE REVIEW

File: fluent/src/commonMain/kotlin/com/konyaco/fluent/component/CalendarView.kt [hunk 1]
```diff
@@ -0,0 +1,681 @@
...
+}
+
+
+@Composable
+private fun PaginationButton(
+    up: Boolean,
+    onClick: () -> Unit
+) {
+    SubtleButton(onClick = onClick, iconOnly = true) {
+        if (up) Icon(
```
Comment (Sanlorng) [inline fluent/src/commonMain/kotlin/com/konyaco/fluent/component/CalendarView.kt:266]: move the statement switch to `imageVector = `, which can reuse the `icon` and `Modifier` component:
```kotlin
Icon(
    imageVector = if (up) Icons.Filled.CareUp else Icons.Filled.CaretDown,
    contentDescrpition = if (up) "Up" else "Down",
    modifier = Modifier.size(12.dp),
)
```

File: fluent/src/commonMain/kotlin/com/konyaco/fluent/component/CalendarView.kt [hunk 2]
```diff
@@ -0,0 +1,681 @@
...
+        Box(
+            modifier = Modifier.clickable(
+                interactionSource = interactionSource,
+                indication = null
+            ) { onSelectedChange(!selected) },
+            contentAlignment = Alignment.Center
+        ) {
+            Box(
+                Modifier.fillMaxSize().padding(1.dp).then(
+                    if (current && selected) Modifier.border(1.dp, Color.Black, CircleShape)
```
Comment (Sanlorng) [inline fluent/src/commonMain/kotlin/com/konyaco/fluent/component/CalendarView.kt:349]: it should keep the same with background color, not the `Color.Black`

**Comment by Sanlorng** [comment]
I see the figma file, unselected item hover backgournd size is `40`  whether it is selected or unselected. but in the current implementation, they are not the same.
figma
figma

gallery

File: fluent/src/commonMain/kotlin/com/konyaco/fluent/component/CalendarView.kt [hunk 3]
```diff
@@ -0,0 +1,681 @@
...
+import com.konyaco.fluent.background.BackgroundSizing
+import com.konyaco.fluent.background.Layer
+import com.konyaco.fluent.component.CalendarDatePickerState.ChooseType
+import com.konyaco.fluent.icons.Icons
+import com.konyaco.fluent.icons.filled.CaretDown
+import com.konyaco.fluent.icons.filled.CaretUp
+import com.konyaco.fluent.icons.regular.CalendarLtr
+import com.konyaco.fluent.scheme.VisualState
+import com.konyaco.fluent.scheme.collectVisualState
+import java.util.Calendar
```
[the remark this answers is not in the document]
Comment (Konyaco) [reply fluent/src/commonMain/kotlin/com/konyaco/fluent/component/CalendarView.kt:51]: I am not sure whether it supports the locale.

[the remark this answers is not in the document]
Comment (Sanlorng) [reply fluent/src/commonMain/kotlin/com/konyaco/fluent/component/CalendarView.kt:51]: Do you mean time zone?

**Comment by Sanlorng** [comment]
We should merge this into dev？and move to kotlinx-datetime when it support LocaleDateFormat or we prepare support for NonJVM target ？

**Comment by Konyaco** [comment]
> We should merge this into dev？and move to kotlinx-datetime when it support LocaleDateFormat or we prepare support for NonJVM target ？

It's better to merge this, we can mark it as not implemented in other targets.

**Pull request merged**